devicetypes/qubino/qubino-dimmer.src/qubino-dimmer.groovy

```diff
@@ -0,0 +1,634 @@
+	def getStatusDelay = duration < 128 ? (duration * 1000) + 2000 : (Math.round(duration / 60) * 60 * 1000) + 2000
+	encapCommands([
+			zwave.switchMultilevelV3.switchMultilevelSet(value: level, dimmingDuration: dimmingDuration),
+			zwave.switchMultilevelV3.switchMultilevelGet()
+	], getStatusDelay)
+}
+
+/**
+ * PING is used by Device-Watch in attempt to reach the Device
+ * */
+def ping() {
+	refresh()
+}
+
+def refresh() {
+	log.debug "refresh"
+	refreshChild()
+	encapCommands(getRefreshCommands())
+}
+
+def getRefreshCommands() {
+	def commands = []
+	commands << zwave.basicV1.basicGet()
+
+	if(isFlushDimmer() || isDINDimmer()) {
+		commands << zwave.meterV2.meterGet(scale: 0)
+		commands << zwave.meterV2.meterGet(scale: 2)
+	}
+	commands
+}
+
+private refreshChild() {
+	// refresh a child temperature sensor (if available)
+	if(childDevices){
+		def childDni = "${device.deviceNetworkId}:2"
+		def childDevice = childDevices.find { it.deviceNetworkId == childDni }
+
+		if (childDevice != null) {
+			sendHubCommand(zwave.sensorMultilevelV5.sensorMultilevelGet(sensorType: 1, scale: 0).format())
+		}
+	}
+}
+
+private encapCommands(commands, delay=200) {
+	delayBetween(commands.collect{ encap(it) }, delay)
+}
+
+private encap(cmd, endpoint = null) {
+	if (cmd) {
+		if (endpoint) {
+			cmd = zwave.multiChannelV3.multiChannelCmdEncap(destinationEndPoint: endpoint).encapsulate(cmd)
+		}
+
+		if (zwaveInfo.zw.endsWith("s")) {
+			zwave.securityV1.securityMessageEncapsulation().encapsulate(cmd).format()
+		} else {
+			cmd.format()
+		}
+	}
+}
+
+private isFlushDimmer(){
+	zwaveInfo.mfr.equals("0159") && zwaveInfo.model.equals("0051")
+}
+
+private isDINDimmer(){
+	zwaveInfo.mfr.equals("0159") && zwaveInfo.model.equals("0052")
+}
+
+private isFlushDimmer010V(){
+	zwaveInfo.mfr.equals("0159") && zwaveInfo.model.equals("0053")
+}
+
+private getParameterMap() {[
+	[
+		name: "Input 1 switch type", key: "input1SwitchType", type: "enum",
+		parameterNumber: 1, size: 1, defaultValue: 0,
+		values: [
+			0: "Default value - Mono-stable switch type (push button) – button quick press turns between previous set dimmer value and zero)",
+			1: "Bi-stable switch type ",
+			2: "Potentiometer (Flush Dimmer 0-10V is using set value the last received from potentiometer or from z-wave controller)",
+			3: "0-10V Temperature sensor (regulated output)"
+		],
+		description: "Set input based on device type (switch, potentiometer, temperature sensor,..)." +
+			"After parameter change to value 3 first exclude module (without setting parameters to default value) then wait at least 30s and then re include the module! "
+	],
+	[
+		name: "Input I1 Sensor reporting", key: "inputI1SensorReporting", type: "range",
+		parameterNumber: 140, size: 2, defaultValue: 5,
+		range: "1..10000",
+		description: "Input I1 Sensor reporting If analogue sensor is connected, module reports measured value on change defined by this parameter. " +
```

Comment:
I'd remove the repetition of the preference name

Reply:
Fixed (that parameter was removed)